pages/static/serviceworker.js

```diff
@@ -9,3 +9,67 @@ AMP_SW.init({
     assets: [],
   },
 });
+
+const SEARCH_CACHE_NAME = 'AMP-DEV-SEARCH-CACHE';
+const SEARCH_LATEST_QUERY_PATH = '/search/latest-query';
+
+// Handle search request
+async function searchDoRequestHandler(url, request) {
+  // Extract query parameter from search request url
+  // Put new query as response to AMP-DEV-SEARCH-CACHE
+  const searchQuery = url.search.match(/q=([^&]+)/)[1];
+  const cache = await caches.open(SEARCH_CACHE_NAME);
+
+  cache.put(SEARCH_LATEST_QUERY_PATH, new Response(`"${searchQuery}"`));
```

Comment:
Should we cache all search queries or wouldn't it be better to clear the cache before adding a new value to avoid stale results?

Reply:
Fixed that. Now the previous result cache will be deleted before a new cache is created.